### PR TITLE
merging DOM cleanup

### PR DESCRIPTION
**Pull request by lmccart**
This PR merges https://github.com/processing/p5.js/pull/3891, renames the file from p5.dom.js to dom.js, updates documentation, and cleans up some tests and other files.

### CODE REVIEW

File: test/unit/addons/p5.dom.js [hunk 1]
```diff
@@ -1,123 +0,0 @@
-/* global testSketchWithPromise */
```
Comment (outofambit) [inline test/unit/addons/p5.dom.js]: @lmccart can we keep these tests? or were they moved elsewhere and i'm just not seeing them?

Comment (lmccart) [reply test/unit/addons/p5.dom.js]: oops good catch, i'll restore them and move to `test/unit/dom`

**Comment by shiffman** [comment]
😍😍😍 👏👏👏

**Comment by Rodion8888** [comment]
https://github.com/outofambit/p5.js/runs/210129233#step:3:1

**Pull request merged**